test/callbacks.js

```diff
@@ -109,6 +126,18 @@ QUnit.module( "module1", {
 		} );
 	} );
 
+	QUnit.module( "module3", {
+		before: callback( "module3 > before" ),
+		beforeEach: callback( "module3 > beforeEach" ),
+		afterEach: callback( "module3 > afterEach" ),
+		after: callback( "module3 > after" )
+	}, function() {
+		QUnit.test( "test1", function( assert ) {
+			invokedHooks.push( "module3 > test1" );
+			assert.ok( true );
```

Comment:
I don't think it is needed. The verification of all the logging should verify the assertion counts as well (since `QUnit.log` is invoked per assertion).